connect/runtime/src/main/java/org/apache/kafka/connect/runtime/standalone/StandaloneHerder.java

```diff
@@ -220,6 +218,11 @@ private String startConnector(Map<String, String> connectorProps) {
             state.configOriginals = connectorProps;
             state.config = connConfig;
         }
+
+        if (tempConnectors.containsKey(connName)) {
```

Comment:
The intention is to remove the temp connectors from the cache if there is an connector with the same name is submitted for execution.